### PR TITLE
Add sensu_subscription module

### PR DESCRIPTION
**Pull request by andsens**
This PR goes well with PR #203.

The module `sensu_subscription` manages sensu subscriptions of the sensu client running on a machine.

### CODE REVIEW

**Comment by gregdek** [comment]
Thanks for submitting this module to Ansible Extras. Apologies that it’s taken a while to get your module reviewed.

To help facilitate reviews, we’ve broadened the number of people who can approve modules for inclusion into the Extras repository. The list of official reviewers can be found here:

https://github.com/ansible/ansible-modules-extras/blob/devel/REVIEWERS.md 

Our new policy is that if a new module is reviewed and approved by at least two official module reviewers, the module will be approved for inclusion.  We will be asking the community of reviewers to take a look at these modules on a regular basis.

To ensure that your module has the best chance of being approved, please double-check that you adhere to the Ansible module guidelines: http://docs.ansible.com/developing_modules.html#module-checklist

**Comment by andsens** [comment]
I went through the checklist and it seems like everything is in order.

**Comment by gregdek** [comment]
Hi monitoring module owners -- @bpennypacker @hkariti @ccollicutt @arturaz @skornehl @bwhaley @abulimov -- anyone interested in helping with another sensu module review? (See #203 for the first one)

**Comment by andsens** [comment]
I applied the comments from #203 to this PR as well.

**Comment by andsens** [comment]
OK. I fixed the errors. Man, python 2.4 compat is a PITA

**Comment by andsens** [comment]
I added some 2.4 compat, but I can see you already merged. You can apply this instead if you like:
https://github.com/andsens/ansible-modules-extras/commit/f33eded84c2f7ff8b7cabfeaadd9c78f19f525e1

**Comment by gregdek** [comment]
Thanks @andsens -- still looking for reviewers for the module itself.

**Comment by andsens** [comment]
Any chance this will be merged soon? It goes well with the `sensu_check` module from PR #203

**Comment by robynbergeron** [comment]
Hi @andsens --  comments below are re: the new process for PRs in extras. We will be evaluating all new module PRs according to this process, effective immediately. If you have any sensu + ansible loving friends, they are welcome to follow the guidelines below and help get this in :)

# 

Thanks for submitting this new module to Ansible Extras! This module is now in community review, a process that is open to all Ansible users. In order for this module to be approved, it must gain the following votes:

“works_for_me”: If you have tested the module thoroughly, including testing of all of the module’s options, and if the module works for you, please add “works_for_me” in the comments.

“passes_guidelines”: If you have gone through the module guidelines and the module meets all of the requirements, please add “passes_guidelines” in the comments. Guidelines are available here: http://docs.ansible.com/developing_modules.html#module-checklist

“needs_revision”: If the module fails to work for you, or if it doesn’t meet guidelines, please add “needs_revision” in the comments with details about what needs to be fixed.

When a module has both “works_for_me” and “passes_guidelines” tags, we will promote the module for inclusion in Ansible Extras.  At this point, you will be expected to maintain the module by fixing bugs and evaluating pull requests in a timely manner. 

Thanks again for submitting your Ansible module!

**Comment by andsens** [comment]
Any chance this can get merged soon?
The sensu_check module is already in 2.0.

works_for_me

**Comment by gregdek** [comment]
Thanks @andsens for this PR. This PR requires revisions, either because it fails to build or by reviewer request. Please make the suggested revisions. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by andsens** [comment]
ready_for_review
The failure is not in the module:

```
$ python2.4 -m compileall -fq -x 'cloud/|monitoring/zabbix.*\.py|/layman\.py|/maven_artifact\.py|clustering/consul.*\.py|notification/pushbullet\.py' .
Compiling ./clustering/znode.py ...
  File "./clustering/znode.py", line 123
    command_type = 'op' if 'op' in module.params and module.params['op'] is not None else 'state'
                         ^
SyntaxError: invalid syntax
```

**Comment by gregdek** [comment]
Yep, unrelated breakage. Should resolve itself when the issue is fixed in the base branch, at which point this _should_ flip back to community_review. @andsens give it a couple of days and we'll see. If not, ping me in the comments.

**Comment by gregdek** [comment]
Thanks @andsens for this PR. This PR requires revisions, either because it fails to build or by reviewer request. Please make the suggested revisions. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@andsens A friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open pending your changes. When you do address the issues, please respond with ready_for_review in your comment, so that we can notify the maintainer.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by andsens** [comment]
ready_for_review

I rebased the branch and added RETURN documentation.

**Comment by gregdek** [comment]
Thanks @andsens for this new module. When this module receives 'shipit' comments from two community members and any 'needs_revision' comments have been resolved, we will mark for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by jaredledvina** [comment]
Nice! I wish this was already in Extras!

shipit

**Comment by gregdek** [comment]
Thanks @andsens for this PR. This PR requires revisions, either because it fails to build or by reviewer request. Please make the suggested revisions. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by andsens** [comment]
aw man, this is getting a little annoying. I'm not bumping the version number _again_. @gregdek, could you merge and bump the version? This module has been fine for almost a year now and keeps breaking because something keeps changing in ansible core.

**Comment by gregdek** [comment]
Apologies @andsens -- aware that the new module review process is not as well integrated as it could be. Will add this module to the agenda of our new module meeting on IRC.

File: monitoring/sensu_subscription.py [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+    name = module.params['name']
+    state = module.params['state']
+    backup = module.params['backup']
+
+    changed, reasons = sensu_subscription(module, path, name, state, backup)
+
+    module.exit_json(path=path, name=name, changed=changed, msg='OK', reasons=reasons)
+
+from ansible.module_utils.basic import *
+main()
```
Comment (gundalow) [inline monitoring/sensu_subscription.py]: This needs to be as follows so debuggers can work with ziploader:

```
if __name__ == '__main__':
    main()
```

**Comment by gregdek** [comment]
@gundalow are these the only changes you see being needed? If so, should we consider a merge and fix?

**Comment by gundalow** [comment]
I'm happy with a merge-and-fix

**Comment by gundalow** [comment]
@bcoca  will address my comments once this has been merged

shipit

**Comment by gregdek** [comment]
Thanks @gundalow -- @andsens sorry for the long delays, and thanks for hanging in there with us.

**Comment by andsens** [comment]
@gregdek, no probs. I'm just glad to see it merged ;-)

**Comment by gregdek** [comment]
Thanks @andsens for this PR. This PR requires revisions, either because it fails to build or by reviewer request. Please make the suggested revisions. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gundalow** [comment]
@bcoca I believe you offered to merge and fix this, for reference the two items that need tweaking are:
- ERROR: version_added should be 2.2. Currently 2.1
- Add `if __name__ == '__main__':`

**Comment by gregdek** [comment]
@andsens A friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open pending your changes. When you do address the issues, please respond with ready_for_review in your comment, so that we can notify the maintainer.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by andsens** [comment]
ready_for_review

Please merge this one before 2.2 gets released :-)

**Comment by gregdek** [comment]
Thanks @andsens for this new module. When this module receives 'shipit' comments from two community members and any 'needs_revision' comments have been resolved, we will mark for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gundalow** [comment]
Shipit

@bcoca I believe you offers to merge this in a previous new modules meeting, could you please do the necessary. Thanks

**Comment by jaredledvina** [comment]
shipit

File: monitoring/sensu_subscription.py [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+            changed = True
+            reasons.append('channel subscription was present and state is `absent\'')
+
+    if changed and not module.check_mode:
+        if backup:
+            module.backup_local(path)
+        try:
+            open(path, 'w').write(json.dumps(config, indent=2) + '\n')
+        except IOError, e:
+            module.fail_json(msg=str(e))
```
Comment (bcoca) [inline monitoring/sensu_subscription.py:133]: we normally want to return a more 'friendly' error message to the user, but leaving that for a follow up (same issue in a couple of places):

`'We could not write to the %s file: %s' % (config, str(e))`

Comment (ghost) [inline monitoring/sensu_subscription.py:133]: Syntax at 132 is breaking jobs

https://github.com/ansible/ansible-modules-extras/pull/1784

**Pull request merged**